hack/jenkins/e2e.sh

```diff
@@ -899,6 +899,7 @@ export KUBERNETES_PROVIDER=${KUBERNETES_PROVIDER}
 export PATH=${PATH}:/usr/local/go/bin
 export KUBE_SKIP_UPDATE=y
 export KUBE_SKIP_CONFIRMATIONS=y
+export HAIRPIN_MODE="${HAIRPIN_MODE:-true}"
```

Comment:
I'd probably opt to leave the default unset so it's only defined in one place.

Which is what @dchen1107 did in #21025 :)